products/BellHybrid/apps/common/src/models/SongsModel.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <common/models/SongsModel.hpp>
+#include <common/widgets/LabelOption.hpp>
+
+namespace app
+{
+    SongsProvider::SongsProvider(ApplicationCommon *app) : DatabaseModel(app)
+    {}
+
+    SongsModel::SongsModel(ApplicationCommon *application,
+                           std::unique_ptr<AbstractSoundsRepository> soundsRepository,
+                           const std::map<std::string, std::string> &pathPrefixes)
+        : SongsProvider(application),
+          application(application), songsRepository{std::move(soundsRepository)}, pathPrefixes{pathPrefixes}
+    {}
+
+    void SongsModel::createData(OnActivateCallback callback)
+    {
+        activateCallback = callback;
+
+        songsRepository->init();
+    }
+
+    void SongsModel::updateRecordsCount()
+    {
+        songsRepository->updateFilesCount();
+    }
+
+    unsigned int SongsModel::requestRecordsCount()
+    {
+        return songsRepository->getFilesCount();
+    }
```

Comment:
Personally I prefer to separate each method with newline - that way it's IMO easier to read

Reply:
Me too! I forgot, done ;)